docs/form.mdx

```diff
@@ -0,0 +1,180 @@
+   id) with the uploaded file, so you can process it on the backend.
+   `@uppy/form` extracts the input values before uploading/processing files and
+   adds them to Uppy meta data state (`uppy.state.meta`), as well as and each
+   file’s meta, and appends to the upload in an object with
+   `[file input name attribute]` -> `[file input value]` key/values.
+2. When you want to first upload the files and put the Uppy response (such as
+   the file URLs) into a hidden field
+   (`<input type="hidden" name="uppyResult">`) to then POST and handle the form
+   yourself. The appended result is a stringified version of a result returned
+   from calling `uppy.upload()` or listening to `complete` event.
+3. When you want to automatically start the file upload on submit or submit the
+   form after Uppy has successfully uploaded. This is off by default. See
+   [`submitOnSuccess`](#submitOnSuccess) and
+   [`triggerUploadOnSubmit`](#triggerUploadOnSubmit) options respectively for
+   details.
+
+:::note
+
+If you are using a UI framework or library like React, Vue or Svelte, you’ll
+most likely handle form data there as well, and thus won’t need this plugin.
+Instead, pass meta data to Uppy via
+[`uppy.setMeta()`](/docs/uppy-core#setmetadata) and listen to
+[`uppy.on('complete')`](/docs/uppy-core#complete) to get the upload results
+back.
+
+:::
+
+## Install
+
+<Tabs>
+  <TabItem value="npm" label="NPM" default>
+
+```shell
+npm install @uppy/form
+```
+
+  </TabItem>
+
+  <TabItem value="yarn" label="Yarn">
+
+```shell
+yarn add @uppy/form
+```
+
+  </TabItem>
+
+  <TabItem value="cdn" label="CDN">
+    <UppyCdnExample>
+      {`
+        import { Uppy, Form } from "{{UPPY_JS_URL}}"
+        const uppy = new Uppy()
+        uppy.use(Form, {
+          // Options
+        })
+      `}
+    </UppyCdnExample>
+  </TabItem>
+</Tabs>
+
+## Use
+
+```js title="app.js"
+import Uppy from '@uppy/core';
+import Form from '@uppy/form';
+import DragDrop from '@uppy/drag-drop';
+
+import '@uppy/core/dist/style.min.css';
+import '@uppy/drag-drop/dist/style.min.css';
+
+new Uppy().use(Form, {
+	target: '#my-form',
+});
+```
+
+```html title="index.html"
+<form id="my-form" action="/submit" method="get">
+	<label for="name">Enter your name: </label>
+	<input type="text" name="name" required />
+
+	<label for="dob">Date of birth: </label>
+	<input type="date" name="dob" />
+
+	<input type="submit" value="Save" />
+</form>
+```
+
+By default the code above will:
+
+1. Extract meta data from the form `#my-form` and send it with the Uppy upload.
+   Those fields will then be added to Uppy meta data state (`uppy.state.meta`)
+   and each file’s meta, and appended as (meta)data to the upload in an object
+   with `[file input name attribute]` -> `[file input value]` key/values.
+2. When Uppy completes upload/processing, it will add an
+   `<input type="hidden" name="uppyResult">` with the stringified upload result
+   object back to the form.
```

Comment:
```suggestion
1. Extract meta data from the form `#my-form`.
2. Send it with the Uppy upload.
3. Those fields will then be added to Uppy meta data state (`uppy.state.meta`)
   and each file’s meta, and appended as (meta)data to the upload in an object
   with `[file input name attribute]` -> `[file input value]` key/values.
4. When Uppy completes upload/processing, it will add an
   `<input type="hidden" name="uppyResult">` with the stringified upload result
   object back to the form.
```